sdk/tables/Azure.Data.Tables/src/TableErrorCode.cs

```diff
@@ -0,0 +1,235 @@
+        private const string TooManyPropertiesValue = "TooManyProperties";
+        private const string EntityTooLargeValue = "EntityTooLarge";
+        private const string PropertyValueTooLargeValue = "PropertyValueTooLarge";
+        private const string KeyValueTooLargeValue = "KeyValueTooLarge";
+        private const string InvalidValueTypeValue = "InvalidValueType";
+        private const string TableBeingDeletedValue = "TableBeingDeleted";
+        private const string PrimaryKeyPropertyIsInvalidTypeValue = "PrimaryKeyPropertyIsInvalidType";
+        private const string PropertyNameTooLongValue = "PropertyNameTooLong";
+        private const string PropertyNameInvalidValue = "PropertyNameInvalid";
+        private const string InvalidDuplicateRowValue = "InvalidDuplicateRow";
+        private const string CommandsInBatchActOnDifferentPartitionsValue = "CommandsInBatchActOnDifferentPartitions";
+        private const string JsonFormatNotSupportedValue = "JsonFormatNotSupported";
+        private const string AtomFormatNotSupportedValue = "AtomFormatNotSupported";
+        private const string JsonVerboseFormatNotSupportedValue = "JsonVerboseFormatNotSupported";
+        private const string MediaTypeNotSupportedValue = "MediaTypeNotSupported";
+        private const string MethodNotAllowedValue = "MethodNotAllowed";
+        private const string ContentLengthExceededValue = "ContentLengthExceeded";
+        private const string AccountIOPSLimitExceededValue = "AccountIOPSLimitExceeded";
+        private const string CannotCreateTableWithIOPSGreaterThanMaxAllowedPerTableValue = "CannotCreateTableWithIOPSGreaterThanMaxAllowedPerTable";
+        private const string PerTableIOPSIncrementLimitReachedValue = "PerTableIOPSIncrementLimitReached";
+        private const string PerTableIOPSDecrementLimitReachedValue = "PerTableIOPSDecrementLimitReached";
+        private const string SettingIOPSForATableInProvisioningNotAllowedValue = "SettingIOPSForATableInProvisioningNotAllowed";
+        private const string PartitionKeyEqualityComparisonExpectedValue = "PartitionKeyEqualityComparisonExpected";
+        private const string PartitionKeySpecifiedMoreThanOnceValue = "PartitionKeySpecifiedMoreThanOnce";
+        private const string InvalidInputValue = "InvalidInput";
+        private const string NotImplementedValue = "NotImplemented";
+        private const string OperationTimedOutValue = "OperationTimedOut";
+        private const string OutOfRangeInputValue = "OutOfRangeInput";
+        private const string ForbiddenValue = "Forbidden";
+
+        /// <summary> AuthorizationResourceTypeMismatch. </summary>
+        public static TableErrorCode AuthorizationResourceTypeMismatch { get; } = new TableErrorCode(AuthorizationResourceTypeMismatchValue);
+
+        /// <summary> XMethodNotUsingPost. </summary>
+        public static TableErrorCode XMethodNotUsingPost { get; } = new TableErrorCode(XMethodNotUsingPostValue);
```

Comment:
The `X` prefix here is an odd pattern.   Is there value to that for customers or would dropping it help provide a more meaningful name?

Reply:
I'm not sure - these come directly from the service, so I'd presume not to change them unless we have good reason to

https://docs.microsoft.com/en-us/rest/api/storageservices/table-service-error-codes